backend/src/datasource/permissions.js

```diff
@@ -0,0 +1,28 @@
+const { rule, shield, allow } = require("graphql-shield");
+
+const isAuthenticatedUser = rule({cache: 'contextual'})(
+  async (parent, args, context) => {
+    let userData = context.userData
+
+    if (context.dataSources.udb.userExists(userData.userId)){
```

Comment:
* Specific to our use-case: Make sure a JWT is only valid if a corresponding user in the database exists. The user might have been deleted or disabled since the